app.rb

```diff
@@ -1,37 +1,36 @@
-require_relative './book'
-require_relative './student'
-require_relative './teacher'
-require_relative './rental'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'people'
+require_relative 'printer'
+require_relative 'bookcreator'
 
 class App
   def initialize
-    @books = []
-    @people = []
-    @rentals = []
-    @num = 0
+    @books = BookList.new
+    @people = People.new
   end
 
   def start
     puts 'Welcome to School Library!'
-    menu while @num != 7
+    menu
   end
 
   private
 
   def menu
-    until @num < 7 && @num.positive?
+    loop do
       puts "Please choose an option by entering a number\n1- List all books\n2- List all people\n3- Create a person
 4- Create a book\n5- Create a rental\n6- List all rentals\n7- exit"
-      @num = gets.chomp.to_i
-    end
-    case @num.to_i
-    when 1 then list_books_flow
-    when 2 then list_people_flow
-    when 3 then add_person
-    when 4 then create_book
-    when 5 then create_rental
-    when 6 then list_rentals
-    when 7 then puts 'exiting'
+      num = gets.chomp.to_i
+      case num.to_i
+      when 1 then ListPrinter.print_list(@books.list)
+      when 2 then ListPrinter.print_list(@people.list)
+      when 3 then add_person
+      when 4 then @books.add_book
+      when 5 then create_rental
+      when 6 then list_rentals
+      when 7 then break
+      end
     end
   end
```

Comment:
- [ ] I have highlighted some of the code that should not belong in this file. Right now this `app.rb` file contains the options for requesting the initial user inputs when the app is run and the other app logic. In keeping with the single-responsibility rule, this file should only contain the logic for the app. The initial requests for user inputs should be put into another file named `main.rb` whose sole responsibility is to start the app and request initial inputs.

  Kindly move these initial user requests and the app startup prompt at the bottom of the file into a different `main.rb` file.